### ISSUE TITLE
Bad result in tfp.distributions.Categorical ?

### ISSUE THREAD

**Issue by wisnunugroho21** [tensorflow/probability#585]
tfp.distributions.Categorical outputs different value with pytorch (torch.distributions.categorical)  
working with tfp version 0.8.0 on Google Colab  

```
import tensorflow as tf
import tensorflow_probability as tfp

a1 = tf.constant([[0.9999, 0.00005, 0.00005], [0.9999, 0.00005, 0.00005], [00.9999, 0.00005, 0.00005]], dtype = tf.float32)
a2 = tf.constant([0, 1, 2], dtype = tf.float32)

distribution = tfp.distributions.Categorical(a1)
log_prob = distribution.log_prob(a2)
probs = distribution.probs_parameter()
entropy = distribution.entropy()
  
print(log_prob)
print(tf.math.exp(log_prob))
print(probs)
print(entropy)
for i in range(10):
    print(distribution.sample())
```

```
tf.Tensor([-0.5515084 -1.5513585 -1.5513585], shape=(3,), dtype=float32)
tf.Tensor([0.57608014 0.21195984 0.21195984], shape=(3,), dtype=float32)
tf.Tensor(
[[0.5760802  0.21195987 0.21195987]
 [0.5760802  0.21195987 0.21195987]
 [0.5760802  0.21195987 0.21195987]], shape=(3, 3), dtype=float32)
tf.Tensor([0.9753646 0.9753646 0.9753646], shape=(3,), dtype=float32)
tf.Tensor([1 1 2], shape=(3,), dtype=int32)
tf.Tensor([1 0 1], shape=(3,), dtype=int32)
tf.Tensor([1 0 2], shape=(3,), dtype=int32)
tf.Tensor([0 1 0], shape=(3,), dtype=int32)
tf.Tensor([1 0 0], shape=(3,), dtype=int32)
tf.Tensor([2 1 1], shape=(3,), dtype=int32)
tf.Tensor([2 0 0], shape=(3,), dtype=int32)
tf.Tensor([1 2 2], shape=(3,), dtype=int32)
tf.Tensor([0 0 0], shape=(3,), dtype=int32)
tf.Tensor([1 1 0], shape=(3,), dtype=int32)
```
```
import torch
import torch.nn as nn
from torch.distributions import Categorical

a1 = torch.tensor([[0.9999, 0.00005, 0.00005], [0.9999, 0.00005, 0.00005], [00.9999, 0.00005, 0.00005]], dtype = torch.float32)
a2 = torch.tensor([0, 1, 2], dtype = torch.float32)

distribution = Categorical(a1)  
log_prob = distribution.log_prob(a2)
probs = distribution.probs
entropy = distribution.entropy()

print(log_prob)
print(log_prob.exp())
print(probs)
print(entropy)
for i in range(10):
    print(distribution.sample())
```
```
tensor([-1.0002e-04, -9.9035e+00, -9.9035e+00])
tensor([9.9990e-01, 5.0000e-05, 5.0000e-05])
tensor([[9.9990e-01, 5.0000e-05, 5.0000e-05],
        [9.9990e-01, 5.0000e-05, 5.0000e-05],
        [9.9990e-01, 5.0000e-05, 5.0000e-05]])
tensor([0.0011, 0.0011, 0.0011])
tensor([0, 0, 0])
tensor([0, 0, 0])
tensor([0, 0, 0])
tensor([0, 0, 0])
tensor([0, 0, 0])
tensor([0, 0, 0])
tensor([0, 0, 0])
tensor([0, 0, 0])
tensor([0, 0, 0])
tensor([0, 0, 0])
```

**Comment by SiegeLordEx** [comment]
TFP `Categorical` takes logits as its first argument, while Torch `Categorical` takes probs as its first argument. To get the same output as Torch, construct the distribution like so: `distribution = tfp.distributions.Categorical(probs=a1)`.
